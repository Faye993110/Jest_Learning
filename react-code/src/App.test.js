import React from 'react';
//import ReactDOM from 'react-dom';
import Enzyme, {shallow, mount}from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from './App';

Enzyme.configure({adapter: new Adapter() });

it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//  // ReactDOM.unmountComponentAtNode(div);
//  const container = div.getElementsByClassName('App');
//  expect(container.length).toBe(1);


// const wrapper = shallow(<App/>);
// const container = wrapper.find('[data-test="container"]');
// expect(container.length).toBe(1);
// expect(container.prop('title')).toBe('Faye');
// expect(container).toExist();
// expect(container).toHaveProp('title', 'Faye');

const wrapper = mount(<App/>);
expect(wrapper).toMatchSnapshot();
});
