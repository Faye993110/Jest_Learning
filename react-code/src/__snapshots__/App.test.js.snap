// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<App>
  <div
    className="app-container"
    data-test="container"
    title="Faye"
  >
    hello world!
  </div>
</App>
`;
